lib/jekyll/frontmatter_defaults.rb

```diff
@@ -30,6 +30,13 @@ def update_deprecated_types(set)
       set
     end
 
+    def ensure_time!(set)
+      return set unless set.key?('values') && set['values'].key?('date')
+      return set if set['values']['date'].is_a?(Time)
+      set['values']['date'] = Utils.parse_date(set['values']['date'], "Front matter defaults does not have a valid date.")
```

Comment:
Can you indicate something more helpful in this error message? If you have a 200-line config.yml, this is going to be an annoying error.